pkg/image/api/helper.go

```diff
@@ -38,18 +49,19 @@ func ParseDockerImageReference(spec string) (DockerImageReference, error) {
 	// TODO replace with docker version once docker/docker PR11109 is merged upstream
 	stream, tag, id := parseRepositoryTag(spec)
 
+	prefix, stream := splitProtocolPrefix(stream)
 	repoParts := strings.Split(stream, "/")
 	switch len(repoParts) {
 	case 2:
 		// namespace/name
-		ref.Namespace = repoParts[0]
+		ref.Namespace = prefix + repoParts[0]
 		ref.Name = repoParts[1]
 		ref.Tag = tag
 		ref.ID = id
 		return ref, nil
 	case 3:
 		// registry/namespace/name
-		ref.Registry = repoParts[0]
+		ref.Registry = prefix + repoParts[0]
```

Comment:
If we do this, we end up with a `dockerImageReference` such as

    dockerImageReference: https://index.docker.io/ncdc/busybox:4986bf8c15363d1c5d15512d5266f8777bfba4974ac56e3270e7760f6f0a8125

This is not a valid pull spec:

    $ docker pull https://index.docker.io/ncdc/busybox:4986bf8c15363d1c5d15512d5266f8777bfba4974ac56e3270e7760f6f0a8125
    FATA[0000] Invalid repository name (ex: "registry.domain.tld/myrepos")